pandas/core/arrays/base.py

```diff
@@ -1236,6 +1241,16 @@ def view(self, dtype: Dtype | None = None) -> ArrayLike:
     # Printing
     # ------------------------------------------------------------------------
 
+    def _values_for_format_object_summary(self) -> np.ndarray:
```

Comment:
we don't want to creat a function each method

rather use the existing ones

Reply:
Oh, okay. I thought about using the _values_for_argsort function, but i think it would be kind of confusing to use it everywhere because of its name. Should i just use it anyways? Or i can create just one function with a general name to use on the other methods.

Reply:
@jbrockmendel do you mind providing a bit more context on what's the exact refactoring you'd like to see in #46843.

I understand your concern, but it's not immediately clear to me what should be done, to help @brunocoutinhoo with this.

Reply:
Thanks for all the info guys. 

I noticed `_rank` passes `self` to `core.algorithms.rank`, which also goes through `_ensure_data`, just like `_mode`. I'm not sure how to move the logic to the EA though, since from what I gather `_ensure_data` is a private function in `core.algorithms`, and it's also being used by other functions inside `core.algorithms`.

Also, in hindsight I'm not exactly sure what should be done in `format_object_summary`, since `io.formats.printing.format_object_summary` expects to receive the object as its first parameter, and from what I understood it doesn't really depend on the EA implementation to work